crates/bazed-core/src/document.rs

```diff
@@ -21,7 +21,17 @@ pub struct Document {
 }
 
 impl Document {
-    pub fn open(path: PathBuf) -> Result<Self> {
+    pub fn open_tmp() -> Result<Self> {
```

Comment:
I don't like tmp as a name for source-less buffers, but we can discuss that another time.